mm2src/coins/solana.rs

```diff
@@ -377,6 +426,311 @@ impl SolanaCoin {
+        ];
+        self.sign_and_send_transaction(swap_program_id, accounts, swap_instruction.pack())
+    }
+
+    fn spend_hash_time_locked_payment(&self, args: SpendPaymentArgs) -> SolTxFut {
+        let sender = Pubkey::new(args.other_pubkey.iter().as_slice());
+        let swap_program_id = Pubkey::new(args.swap_contract_address.as_ref().unwrap().as_slice());
+        let secret: [u8; 32] = <[u8; 32]>::try_from(args.secret).unwrap();
+        let secret_hash = sha256(secret.as_slice());
+        let (lock_time, _secret_hash, amount, token_program) =
+            self.get_transaction_details(args.other_payment_tx).unwrap();
+        let (vault_pda, vault_pda_data, vault_bump_seed, vault_bump_seed_data, _rent_exemption_lamports) =
+            self.create_vaults(lock_time, secret_hash.take(), swap_program_id, 41);
+        let swap_instruction = AtomicSwapInstruction::ReceiverSpend {
+            secret,
+            lock_time,
+            amount,
+            sender,
+            token_program,
+            vault_bump_seed,
+            vault_bump_seed_data,
+        };
+        let accounts = vec![
+            AccountMeta::new(self.key_pair.pubkey(), true), // Marked as signer
+            AccountMeta::new(vault_pda_data, false),        // Not a signer
+            AccountMeta::new(vault_pda, false),             // Not a signer
+            AccountMeta::new(solana_program::system_program::id(), false), //system_program must be included
+        ];
+        self.sign_and_send_transaction(swap_program_id, accounts, swap_instruction.pack())
+    }
+
+    fn refund_hash_time_locked_payment(&self, args: RefundPaymentArgs) -> SolTxFut {
+        let receiver = Pubkey::new(args.other_pubkey.iter().as_slice());
+        let swap_program_id = Pubkey::new(args.swap_contract_address.as_ref().unwrap().as_slice());
+        let (lock_time, secret_hash, amount, token_program) = self.get_transaction_details(args.payment_tx).unwrap();
+        let (vault_pda, vault_pda_data, vault_bump_seed, vault_bump_seed_data, _rent_exemption_lamports) =
+            self.create_vaults(lock_time, secret_hash, swap_program_id, 41);
+        let swap_instruction = AtomicSwapInstruction::SenderRefund {
+            secret_hash,
+            lock_time,
+            amount,
+            receiver,
+            token_program,
+            vault_bump_seed,
+            vault_bump_seed_data,
+        };
+        let accounts = vec![
+            AccountMeta::new(self.key_pair.pubkey(), true), // Marked as signer
+            AccountMeta::new(vault_pda_data, false),        // Not a signer
+            AccountMeta::new(vault_pda, false),             // Not a signer
+            AccountMeta::new(solana_program::system_program::id(), false), //system_program must be included
+        ];
+        self.sign_and_send_transaction(swap_program_id, accounts, swap_instruction.pack())
+    }
+
+    fn get_transaction_details(
+        &self,
+        signature_bytes: &[u8],
```

Comment:
we call this the transaction hash or `tx_hash`. I understand that `signature_bytes` is the tx hash for solana but it's good to have one name across all coins.

Reply:
done